ocaml/gencert/selfcert.ml

```diff
@@ -93,21 +95,25 @@ let selfsign issuer extensions key_length days certfile =
           x
     with e ->
       let msg =
-        Printf.sprintf "generating RSA key for %s failed: %s" certfile
-          (Printexc.to_string e)
+        Printf.sprintf "generating RSA key failed: %s" (Printexc.to_string e)
       in
       D.error "%s" msg ; failwith msg
```

Comment:
This is a bit weird mix of result monad chaining, and exceptions.
*If* we wanted we could use the `Rresult` module, which has a `trap_exn` function that captures an exception together with its backtrace in the `Error` variant.
https://erratique.ch/software/rresult/doc/Rresult.R.html

Reply:
I switched to just using the result monad. I think it's easier to not use `trap_exn` because the debug library in xcp-idl expects `Backtrace.t` for logging, whereas `trap_exn` gives us `Printexn.raw_backtrace` (it's not clear to me that we can convert between the two)